core/auth/src/main/java/software/amazon/awssdk/auth/signer/internal/util/SignerMethodResolver.java

```diff
@@ -56,7 +57,7 @@ public static SigningMethod resolveSigningMethodUsed(Signer signer, ExecutionAtt
                 signingMethod = SigningMethod.PROTOCOL_STREAMING_SIGNING_AUTH;
             } else if (isProtocolBasedUnsigned(signer, executionAttributes)) {
                 signingMethod = SigningMethod.PROTOCOL_BASED_UNSIGNED;
-            } else if (isAnonymous(credentials)) {
+            } else if (isAnonymous(credentials) || signer instanceof NoOpSigner) {
```

Comment:
Are we planning to make NoOpSigner as final class ?

Reply:
Yeah, I think we should do it. I'll create a PR against mainline

Reply:
https://github.com/aws/aws-sdk-java-v2/pull/3481 is submitted to make it a final class